src/components/logs/Context.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+    createContext,
+    useContext,
+    useEffect,
+    useCallback,
+    useState,
+    useRef,
+} from 'react';
+import { BaseComponentProps, ViewLogs_Line } from 'types';
+import { DEFAULT_POLLING_INTERVAL, RPCS } from 'services/supabase';
+import { hasLength, incrementInterval, timeoutCleanUp } from 'utils/misc-utils';
+import { useClient } from 'hooks/supabase-swr';
+
+interface Props extends BaseComponentProps {
+    token: string | null;
+    disableIntervalFetching?: boolean;
+    fetchAll?: boolean;
+}
+
+type LogsContextValue = {
+    reset: () => void;
+    logs: ViewLogs_Line[];
+    networkFailure: boolean;
+    stopped: boolean;
+};
+
+const MAX_EMPTY_CALLS = 90; // about 1.5 minute
+const MAX_INTERVAL = 1000; //  we don't want to wait too longer between pings
+const LogsContext = createContext<LogsContextValue | null>(null);
+
+const LogsContextProvider = ({
+    children,
+    token,
+    disableIntervalFetching,
+    fetchAll,
+}: Props) => {
+    const supabaseClient = useClient();
+
+    // Privates
+    const allLogs = useRef<ViewLogs_Line[]>([]);
+    const offset = useRef(0);
+    const timeoutRef = useRef<number | undefined>();
+    const interval = useRef(DEFAULT_POLLING_INTERVAL / 2);
+    const emptyResponses = useRef(
+        disableIntervalFetching ? MAX_EMPTY_CALLS : 0
+    );
+
+    // Publics
+    const [logs, setLogs] = useState<ViewLogs_Line[]>([]);
+    const [stopped, setStopped] = useState(false);
+    const [networkFailure, setNetworkFailure] = useState(false);
+
+    const fetchLogs = async (offsetVal: number) => {
+        const queryParams = {
+            bearer_token: token,
+        };
+        if (fetchAll) {
+            return supabaseClient
+                .rpc<ViewLogs_Line>(RPCS.VIEW_LOGS, queryParams)
+                .throwOnError();
+        } else {
+            return supabaseClient
+                .rpc<ViewLogs_Line>(RPCS.VIEW_LOGS, queryParams)
+                .throwOnError()
+                .range(offsetVal, offsetVal + 10);
+        }
+    };
+
+    const stopLogs = (networkFailed?: boolean) => {
+        timeoutCleanUp(timeoutRef.current);
+        setStopped(true);
+        emptyResponses.current = emptyResponses.current + 1;
+        if (networkFailed) {
+            setNetworkFailure(true);
+        }
+    };
+
+    const start = () => {
+        if (!token) return;
+
+        const makeApiCall = () => {
```

Comment:
Basically the same as the old one. However, now the max calls is checked within the `then`